### ISSUE TITLE
SIGABRT Crash -> std::_fl::__assoc_sub_state::__execute (new.cpp:44)

### ISSUE THREAD

**Issue by jamalalayeq** [flutter/flutter#155292]
### Steps to reproduce

Our users faced a crash on their Android devices.
[Stack Trace](https://beyondmenu.sentry.io/share/issue/8c4d91ea4045455590a76e190d42e48e/)
Devices are Pixels family.
Flutter version is 3.22.0

### Logs

<details>
<summary>Logs </summary>

```
console
system
exception
stack-trace
frame
frame (ignored due to recursion)
frame
filenamethread.cc
functionfml::ThreadHandle::ThreadHandle::lambda::__invoke
frame (ignored by stack trace rule (category:indirection -group))
filenamethread.cc
functionfml::ThreadHandle::ThreadHandle::lambda::operator()
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::function<T>::operator()
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::__function::__func<T>::operator()
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::__function::__alloc_func<T>::operator()[abi:v15000]
frame (ignored by stack trace rule (category:internals -group))
filenameinvoke.h
functionstd::_fl::__invoke_void_return_wrapper<T>::__call<T>
frame (ignored by stack trace rule (category:internals -group))
filenameinvoke.h
functionstd::_fl::__invoke[abi:v15000]<T>
frame (ignored by stack trace rule (category:indirection -group))
filenamethread.cc
functionfml::Thread::Thread::lambda::operator()
frame
filenamemessage_loop.cc
functionfml::MessageLoop::Run
frame
filenamemessage_loop_impl.cc
functionfml::MessageLoopImpl::DoRun
frame
filenamemessage_loop_android.cc
functionfml::MessageLoopAndroid::Run

collapse 9 similar
frame
functionALooper_pollOnce
frame (ignored by stack trace rule (category:internals -group))
functionandroid::Looper::pollOnce
frame
filenamemessage_loop_android.cc
functionfml::MessageLoopAndroid::MessageLoopAndroid::lambda::__invoke
frame (ignored by stack trace rule (category:indirection -group))
filenamemessage_loop_android.cc
functionfml::MessageLoopAndroid::MessageLoopAndroid::lambda::operator()
frame
filenamemessage_loop_android.cc
functionfml::MessageLoopAndroid::OnEventFired
frame
filenamemessage_loop_impl.cc
functionfml::MessageLoopImpl::RunExpiredTasksNow
frame
filenamemessage_loop_impl.cc
functionfml::MessageLoopImpl::FlushTasks
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::function<T>::operator()
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::__function::__func<T>::operator()
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::__function::__alloc_func<T>::operator()[abi:v15000]
frame (ignored by stack trace rule (category:internals -group))
filenameinvoke.h
functionstd::_fl::__invoke_void_return_wrapper<T>::__call<T>
frame (ignored by stack trace rule (category:internals -group))
filenameinvoke.h
functionstd::_fl::__invoke[abi:v15000]<T>
frame (ignored by stack trace rule (category:indirection -group))
filenamevsync_waiter.cc
functionflutter::VsyncWaiter::FireCallback::lambda::operator()
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::function<T>::operator()
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::__function::__value_func<T>::operator()[abi:v15000]
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::__function::__func<T>::operator()
frame (ignored by stack trace rule (category:indirection -group))
filenamefunction.h
functionstd::_fl::__function::__alloc_func<T>::operator()[abi:v15000]
frame (ignored by stack trace rule (category:internals -group))
filenameinvoke.h
functionstd::_fl::__invoke_void_return_wrapper<T>::__call<T>
frame (ignored by stack trace rule (category:internals -group))
filenameinvoke.h
functionstd::_fl::__invoke[abi:v15000]<T>
frame (ignored by stack trace rule (category:indirection -group))
filenameanimator.cc
functionflutter::Animator::AwaitVSync::lambda::operator()
frame
filenameanimator.cc
functionflutter::Animator::BeginFrame
frame
filenameshell.cc
functionflutter::Shell::OnAnimatorBeginFrame
frame
filenameengine.cc
functionflutter::Engine::BeginFrame
frame
filenameruntime_controller.cc
functionflutter::RuntimeController::BeginFrame
frame
filenameplatform_configuration.cc
functionflutter::PlatformConfiguration::BeginFrame
frame
filenamedart_invoke.cc
functiontonic::DartInvokeVoid
frame
filenamedart_api_impl.cc
functionDart_InvokeClosure
frame
filenamedart_entry.cc
functiondart::DartEntry::InvokeFunction

collapse 24 similar
frame
function_kDartVmSnapshotInstructions
frame
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions
frame (ignored due to recursion)
function_kDartIsolateSnapshotInstructions

collapse 24 similar
frame
filenamedl_builder.cc
functionflutter::DisplayListBuilder::Restore
frame (ignored by stack trace rule (category:internals -group))
filenamenew.cpp
functionstd::_fl::__assoc_sub_state::__execute
frame
functionabort
type (ignored because exception is synthetic)SIGABRT
value (ignored because stacktrace takes precedence)Abort

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.24.1, on macOS 14.6.1 23G93 darwin-arm64, locale en-EG)
    • Flutter version 3.24.1 on channel stable at /Users/jamalalaayq/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5874a72aa4 (4 weeks ago), 2024-08-20 16:46:00 -0500
    • Engine revision c9b9d5780d
    • Dart version 3.5.1
    • DevTools version 2.37.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/jamalalaayq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/jamalalaayq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] VS Code (version 1.93.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.96.0

[✓] Connected device (4 available)
    • Jamal’s iPhone (mobile)         • 00008110-0001305C0162401E • ios            • iOS 17.6.1 21G93
    • macOS (desktop)                 • macos                     • darwin-arm64   • macOS 14.6.1 23G93 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad     • darwin         • macOS 14.6.1 23G93 darwin-arm64
    • Chrome (web)                    • chrome                    • web-javascript • Google Chrome 128.0.6613.138

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
> functionflutter::DisplayListBuilder::Restore

@jamalalayeq A similar issue has been fixed recently https://github.com/flutter/flutter/issues/149305. Since you mentioned the crash is occuring in 3.22, above fix was pushed in latest stable version 3.24. So please try to upgrade to latest stable and then check if you still get same crash or not.

**Comment by jamalalayeq** [comment]
> > functionflutter::DisplayListBuilder::Restore
> 
> @jamalalayeq A similar issue has been fixed recently #149305. Since you mentioned the crash is occuring in 3.22, above fix was pushed in latest stable version 3.24. So please try to upgrade to latest stable and then check if you still get same crash or not.

Thanks, we will do that and inform you of the results.
